### PR TITLE
CPM-908: Add handler to generate family codes

### PR DESCRIPTION
**Pull request by pierallard**

### CODE REVIEW

File: components/identifier-generator/back/tests/Specification/Application/Generate/Property/GenerateFamilyHandlerSpec.php [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        ]);
+
+        $this->shouldThrow(new UnableToTruncateException('AKN-fam', 'sku', 'fam'))->during(
+            '__invoke', [
+                $family,
+                $target,
+                $this->getProductProjection('fam'),
+                'AKN-'
+            ]
+        );
```
Comment (mmetayer) [inline components/identifier-generator/back/tests/Specification/Application/Generate/Property/GenerateFamilyHandlerSpec.php:102]:
```suggestion
        $this->__invoke(
            $family,
            $target,
            $this->getProductProjection('familyCode'),
            'AKN-'
        )->shouldReturn('fam');
```

File: components/identifier-generator/back/tests/Specification/Application/Generate/Property/GenerateFamilyHandlerSpec.php [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        ]);
+
+        $this->shouldNotThrow(UnableToTruncateException::class)->during(
+            '__invoke', [
+                $family,
+                $target,
+                $this->getProductProjection('fam'),
+                'AKN-'
+            ]
+        );
```
Comment (mmetayer) [inline components/identifier-generator/back/tests/Specification/Application/Generate/Property/GenerateFamilyHandlerSpec.php:124]:
```suggestion
        $this->__invoke(
            $family,
            $target,
            $this->getProductProjection('familyCode'),
            'AKN-'
        )->shouldReturn('fam');
```

**Review by mmetayer — APPROVED** [review]

File: components/identifier-generator/back/tests/Specification/Application/Generate/Property/GenerateFamilyHandlerSpec.php (hunk 1 above)
Comment (pierallard) [reply components/identifier-generator/back/tests/Specification/Application/Generate/Property/GenerateFamilyHandlerSpec.php:102]: No ? Why ?

Comment (mmetayer) [reply components/identifier-generator/back/tests/Specification/Application/Generate/Property/GenerateFamilyHandlerSpec.php:102]: arf, sorry, wrong test :weary:

File: components/identifier-generator/back/tests/Specification/Application/Generate/Property/GenerateFamilyHandlerSpec.php [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    }
+
+    public function it_should_not_throw_an_error_if_family_code_is_exactly_the_right_length(): void
```
Comment (mmetayer) [inline components/identifier-generator/back/tests/Specification/Application/Generate/Property/GenerateFamilyHandlerSpec.php:105]: you could also add a test with truncate LTE and a short family code

**Review by Beninho — APPROVED** [review]

**Pull request merged**